lib/schedule.js

```diff
@@ -508,14 +523,14 @@ function cancelInvocation(invocation) {
 /* Recurrence scheduler */
 function scheduleNextRecurrence(rule, job, prevDate, endDate) {
 
-  prevDate = (prevDate instanceof Date) ? prevDate : (new Date());
+  prevDate = (prevDate instanceof Date || prevDate instanceof CronDate) ? prevDate : (new CronDate(undefined, rule.timezone || null));
 
   var date = (rule instanceof RecurrenceRule) ? rule.nextInvocationDate(prevDate) : rule.next();
   if (date === null) {
     return null;
   }
 
-  if ((endDate instanceof Date) && date.getTime() > endDate.getTime()) {
+  if ((endDate instanceof Date || endDate instanceof CronDate) && date.getTime() > endDate.getTime()) {
```

Comment:
And probably here if we make sure of transforming endDate to CronDate in the `Job.schedule` function